subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/ImmutableCapability.java

```diff
@@ -13,103 +13,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.gradle.internal.component.external.model;
 
-import com.google.common.base.Objects;
-import org.gradle.api.artifacts.ModuleVersionIdentifier;
-import org.gradle.api.capabilities.Capability;
+package org.gradle.internal.component.external.model;
 
 import javax.annotation.Nullable;
 
-public final class ImmutableCapability implements CapabilityInternal {
-
-    public static ImmutableCapability defaultCapabilityForComponent(ModuleVersionIdentifier identifier) {
-        return new ImmutableCapability(identifier.getGroup(), identifier.getName(), identifier.getVersion());
-    }
-
-    private final String group;
-    private final String name;
-    private final String version;
-    private final int hashCode;
-    private final String cachedId;
-
+/**
+ * This is a deprecated internal class that exists specifically to be used by the
+ * Android and Kotlin Androind plugins, which expect this type.
```

Comment:
Do you know where in the KGP/AGP plugins this class is being used?

Reply:
I temporarily removed the type and am rerunning tests now.  I'll add that info to the type and restore it after I see what specifically fails.

Reply:
The [failures](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/61405441?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildTestsSection=true) are:

- `KotlinPluginAndroidKotlinDSLSmokeTest` and `KotlinPluginAndroidGroovyDSLSmokeTest` with kotlin 1.6.21, agp 7.3.1 - 8.0.0-alpha11.
- `AndroidPluginsSmokeTest` with 7.3.1 - 8.0.0-alpha11.
- `AndroidSantaTrackerIncrementalCompilationSmokeTest`, `AndroidSantaTrackerCachingCompilationSmokeTest`, `AndroidSantaTrackerDeprecationSmokeTest` with same versions.

They all fail with some version (different configurations and task names) of:

```
  > Could not resolve all dependencies for configuration ':cityquiz:debugRuntimeClasspath'.
     > Could not create task ':santa-tracker:generateDebugFeatureTransitiveDeps'.
        > org/gradle/internal/component/external/model/ImmutableCapability
```

Also some android samples fail in the same manner. 

The task at fault is `com.android.build.gradle.internal.tasks.featuresplit.PackagedDependenciesWriterTask`.

Reply:
could you add this into a comment here